project/BuildPlugin.scala

```diff
@@ -222,27 +227,66 @@ object BuildImplementation {
          |  IO.write(file("$testDir/bloop-config/base-directory"), dir.getAbsolutePath)
          |}
          |TaskKey[Unit]("checkInstall") := {
-         |  Thread.sleep(1000) // Let's wait a little bit because of OS's IO latency
-         |  val mostRecentStamp = (bloopConfigDir.value ** "*.config").get.map(_.lastModified).min
-         |  (System.currentTimeMillis() - mostRecentStamp)
-         |  val diff = (System.currentTimeMillis() - mostRecentStamp) / 1000
-         |  if (diff <= 15) () // If it happened in the last 15 seconds, this is ok!
-         |  else sys.error("The sbt plugin didn't write any file")
+         |  val configFiles = (bloopConfigDir.value ** "*.config").get
+         |  if (configFiles.isEmpty) {
+         |    sys.error("The sbt plugin didn't write any file")
+         |  }
          |}
     """.stripMargin
     }
 
+    private val testPluginContents = {
+      """import sbt._
+        |import Keys._
+        |import bloop.SbtBloop.autoImport._
+        |
+        |object TestPlugin extends AutoPlugin {
+        |  override def requires = bloop.SbtBloop
+        |  override def trigger = allRequirements
+        |
+        |  object autoImport {
+        |    lazy val copySources = taskKey[Unit]("Copy all generated sources")
+        |    lazy val copySourcesIndividual = taskKey[Unit]("Copy generated sources")
+        |  }
+        |  import autoImport._
+        |
+        |  override def globalSettings: Seq[Setting[_]] = Seq(
+        |    copySources := Def.taskDyn {
+        |      val filter = ScopeFilter(sbt.inAnyProject)
+        |      copySourcesIndividual.all(filter).map(_ => ())
+        |    }.value
+        |  )
+        |
+        |  override def projectSettings: Seq[Setting[_]] = Seq(
+        |    copySourcesIndividual := {
+        |      val outOfScriptedRoot = bloopConfigDir.value.getParentFile
+        |      val inScriptedRoot = baseDirectory.in(ThisBuild).value
+        |      def copy(managedSources: Seq[File]): Unit = {
+        |        for { src <- managedSources
+        |              generatedPath <- IO.relativize(inScriptedRoot, src)
+        |              generatedOutOfScripted = outOfScriptedRoot / generatedPath } {
+        |          IO.copyFile(src, generatedOutOfScripted)
+        |        }
+        |      }
+        |      val allManagedSources = (managedSources in Compile).value ++ (managedSources in Test).value
+        |      copy(allManagedSources)
+        |    }
+        |  )
+        |}""".stripMargin
+    }
+
     private val scriptedTestContents = {
       """> show bloopConfigDir
         |> registerDirectory
         |> installBloop
+        |> copySources
         |> checkInstall
       """.stripMargin
     }
 
     private val NewLine = System.getProperty("line.separator")
     def scriptedSettings(testDirectory: sbt.SettingKey[File]): Seq[Def.Setting[_]] = List(
-      ScriptedKeys.scriptedBufferLog := false,
```

Comment:
Are we sure we want this?

Reply:
Drone was killing the build after we hit a limit in the size of the log.